apps/user-profile/src/components/modules/OnboardingLandingPage/OnboardingLandingPage.js

```diff
@@ -1,272 +1,385 @@
+        } else {
+          alert('The file size must be less than 200MB.');
         }
+      } else {
+        alert('Please select a valid image file (PNG or JPEG).');
+      }
     }
+  }
 
-    function onFirstNameChange(e) {
-        console.log(person);
-        setPerson({ ...person, firstName: e.target.value, firstNameTouched: true });
-        console.log("VALIDATION: ", formValidation)
-    }
+  function onFirstNameChange(e) {
+    console.log(person);
+    setPerson({ ...person, firstName: e.target.value, firstNameTouched: true });
+    console.log('VALIDATION: ', formValidation);
+  }
 
-    function onLastNameChange(e) {
-        console.log(person);
-        setPerson({ ...person, lastName: e.target.value, lastNameTouched: true });
-        console.log("VALIDATION: ", formValidation)
-    }
+  function onLastNameChange(e) {
+    console.log(person);
+    setPerson({ ...person, lastName: e.target.value, lastNameTouched: true });
+    console.log('VALIDATION: ', formValidation);
+  }
 
-    function onLocationChange(e) {
-        console.log(person);
-        setPerson({ ...person, location: e.target.value })
-    }
+  function onLocationChange(e) {
+    console.log(person);
+    setPerson({ ...person, location: e.target.value });
+  }
 
-    function onRoleChange(e) {
-        setPerson({ ...person, role: e.target.value });
-        console.log(person);
-    }
+  function onRoleChange(e) {
+    setPerson({ ...person, role: e.target.value });
+    console.log(person);
+  }
 
-    function onHeadLineChange(e) {
-        setPerson({ ...person, role: e.target.value });
-        console.log(person);
-    }
+  function onHeadLineChange(e) {
+    setPerson({ ...person, role: e.target.value });
+    console.log(person);
+  }
 
-    function onTermsAndConditionChange(e) {
-        console.log(person);
-        setPerson({ ...person, termsAndConditions: !person.termsAndConditions });
-    }
-    function onJoinNewsLetterChange() {
-        console.log("Join News")
-    }
+  function onTermsAndConditionChange(e) {
+    console.log(person);
+    setPerson({ ...person, termsAndConditions: !person.termsAndConditions });
+  }
+  function onJoinNewsLetterChange() {
+    console.log('Join News');
+  }
 
-    const onContinueClick = (e) => {
-        router.push({
-            pathname: '/users/me',
-            query: { onboarding: true }
-        }, '/users/me');
-        e.preventDefault()
-    }
+  const onContinueClick = (e) => {
+    const profileId = userData?.profile?.id;
 
-    const onCancelClick = (e) => {
-        router.push({
-            pathname: '/',
-        }, '/');
-        e.preventDefault()
-    }
+    const requestBody = {
+      data: {
+        displayName: `${person.firstName} ${person.lastName}`,
+        profilePictureUrl: uploadedProfilePicture || profilePicture,
+      },
+    };
+    profileId &&
+      axios
+        .put(
+          `${process.env.NEXT_PUBLIC_STRAPI_URL}/profiles/${profileId}/`,
```

Comment:
@Enjoy2Live I think we have a library to send requests to the backend?

Reply:
yes, please use the agent as follows

```
import agent from "@devlaunchers/utility/agent"

agent.profiles.put(profileId, requestBody).then(etc....)
```

keep in mind that agent is exported from `packages/utility/agent.ts` so if the object "profiles" can't be found (seems to be the case rn) you need to create it in that file and export it so you can access `agent.profiles` from within `apps/user-profile/src/components/modules/OnboardingLandingPage/OnboardingLandingPage.js`